epochalyst/_core/_caching/_cacher.py

```diff
@@ -1,46 +1,70 @@
 import glob
+import os
 import pickle
-from epochalyst._core._logging._logger import _Logger
-from typing import Any
+from typing import Any, TypedDict, Literal
+
 import dask.array as da
 import dask.dataframe as dd
 import numpy as np
-import os
 import pandas as pd
+import polars as pl
+
+from epochalyst._core._logging._logger import _Logger
 
 
-class _Cacher(_Logger):
-    """The cacher is a flexible class that allows for caching of any data.
+class _CacheArgs(TypedDict):
+    """The cache arguments.
 
-    The cacher uses cache_args to determine if the data is already cached and if so, return the cached data.
-    cache_args is a dictionary that contains the arguments to determine if the data is already cached. Currently listed cache_args are
-    supported if more are required create a new issue on the github repository.
+    Currently listed cache_args are supported. If more are required, create a new GitHub issue.
 
-    cache_args supports the following keys:
+    The following keys are supported:
         - output_data_type: The type of the output data.
-            - "dask_array": The output data is a dask array.
-            - "numpy_array": The output data is a numpy array.
-            - "pandas_dataframe": The output data is a pandas dataframe.
-            - "dask_dataframe": The output data is a dask dataframe.
+            - "dask_array": The output data is a Dask array.
+            - "numpy_array": The output data is a NumPy array.
+            - "pandas_dataframe": The output data is a Pandas dataframe.
+            - "dask_dataframe": The output data is a Dask dataframe.
+            - "polars_dataframe": The output data is a Polars dataframe.
         - storage_type: The type of the storage.
-            - ".npy": The storage type is a numpy file.
-            - ".parquet": The storage type is a parquet file.
-            - ".csv": The storage type is a csv file.
-            - ".npy_stack": The storage type is a numpy stack.
+            - ".npy": The storage type is a NumPy file.
+            - ".parquet": The storage type is a Parquet file.
+            - ".csv": The storage type is a CSV file.
+            - ".npy_stack": The storage type is a NumPy stack.
             - ".pkl": The storage type is a pickle file
         - storage_path: The path to the storage.
 
+    :param output_data_type: The type of the output data.
+    :param storage_type: The type of the storage.
+    :param storage_path: The path to the storage.
+    """
+
+    output_data_type: Literal[
+        "dask_array",
+        "numpy_array",
+        "pandas_dataframe",
+        "dask_dataframe",
+        "polars_dataframe",
+    ]
+    storage_type: Literal[".npy", ".parquet", ".csv", ".npy_stack", ".pkl"]
+    storage_path: str  # TODO(Jeffrey) Allow str | bytes | os.PathLike[str] | os.PathLike[bytes] instead of just str
```

Comment:
Turn this into an issue instead of a TODO, I feel like TODO's just get lost in the codebase

Reply:
I just created #97 for this.